tests/js/client/aql/aql-optimizer-indexes-group-sort.js

```diff
@@ -191,24 +191,174 @@ function optimizerIndexesGroupSortTestSuite() {
+        [2, "FOR doc IN @@collection SORT doc.a ASC, doc.b ASC, doc.x ASC RETURN doc.b"],
+
+        [1, "FOR doc IN @@collection SORT doc.a DESC, doc.x ASC RETURN doc.b"],
+        [1, "FOR doc IN @@collection SORT doc.a ASC, doc.x DESC RETURN doc.b"],
+        [1, "FOR doc IN @@collection SORT doc.a ASC, doc.b DESC, doc.x DESC RETURN doc.b"],
+        [1, "FOR doc IN @@collection SORT doc.a ASC, doc.b DESC, doc.x ASC RETURN doc.b"],
+        [1, "FOR doc IN @@collection SORT doc.a DESC, doc.b ASC, doc.x DESC RETURN doc.b"],
+        [1, "FOR doc IN @@collection SORT doc.a DESC, doc.b ASC, doc.x ASC RETURN doc.b"],
+        [2, "FOR doc IN @@collection SORT doc.a DESC, doc.b DESC, doc.c ASC, doc.x ASC RETURN doc.b"],
+        [2, "FOR doc IN @@collection SORT doc.a DESC, doc.b DESC, doc.c ASC, doc.x DESC RETURN doc.b"],
+        [2, "FOR doc IN @@collection SORT doc.a ASC, doc.b ASC, doc.c DESC, doc.x ASC RETURN doc.b"],
+        [2, "FOR doc IN @@collection SORT doc.a ASC, doc.b ASC, doc.c DESC, doc.x DESC RETURN doc.b"],
+      ];
+      for (let [numberOfGroupedElements, query] of queries) {
+        let plan = query_plan(query, collection.name());
+        assertTrue(query_plan_uses_index_for_sorting(plan), plan.rules);
+        assertTrue(sort_node_does_a_group_sort(plan, numberOfGroupedElements), plan.nodes);
+        assertEqual(expected_results(query, collection.name()), execute(query, collection.name()), query);
+      }
+
+      queries = [
+        "FOR doc IN @@collection SORT doc.a ASC, doc.b ASC, doc.c ASC, doc.x ASC RETURN doc.b",
+        "FOR doc IN @@collection SORT doc.a ASC, doc.b ASC, doc.c ASC, doc.x DESC RETURN doc.b",
+        "FOR doc IN @@collection SORT doc.a DESC, doc.b DESC, doc.c DESC, doc.x DESC RETURN doc.b",
+        "FOR doc IN @@collection SORT doc.a DESC, doc.b DESC, doc.c DESC, doc.x ASC RETURN doc.b",
+      ];
+      for (let query of queries) {
+        let plan = query_plan(query, collection.name());
+        assertTrue(query_plan_uses_index_for_sorting(plan), plan.rules);
+        // uses full index, therefore full sort node is exchanged by index node
+        assertFalse(sort_node_does_a_group_sort(plan), plan.nodes);
+        assertEqual(expected_results(query, collection.name()), execute(query, collection.name()), query);
+      }
+    },
+
+    ///////////////////////////////////////////////////////////////////////////////
+    /// @brief test index usage
+    ////////////////////////////////////////////////////////////////////////////////
+
+    test_uses_sparse_index_when_sort_registers_start_with_index_fields_and_are_not_null: function () {
+      let randomNumber = randomNumberGeneratorInt(seed);
+      const collection = create_collection();
+      collection.insert(Array.from({ length: 100 }, (_, index) => index).map(i => {
+        return {
+          a: i % 9,
+          x: 100 - i - 1,
+          b: Math.floor(randomNumber() * 100),
+          c: i
+        };
+      }));
```

Comment:
Since the index is sparse, I'd suggest we also have some documents with `null` values in the mix.